m2cgen/assemblers/xgboost.py

```diff
@@ -0,0 +1,106 @@
+import json
+import numpy as np
+from m2cgen import ast
+from m2cgen.assemblers import utils
+from m2cgen.assemblers.base import ModelAssembler
+
+
+class XGBoostModelAssembler(ModelAssembler):
+
+    def __init__(self, model):
+        super().__init__(model)
+        self._base_score = self.model.base_score
+        self._n_estimators = self.model.n_estimators
+
+        feature_names = enumerate(self.model.get_booster().feature_names)
+        self._feature_name_to_idx = {name: idx for idx, name in feature_names}
+
+        self._output_size = 1
+        self._is_classification = False
+        model_class_name = type(model).__name__
+        if model_class_name == "XGBClassifier":
+            self._is_classification = True
+            if self.model.n_classes_ > 2:
+                self._output_size = self.model.n_classes_
+
+    def assemble(self):
+        model_dump = self.model.get_booster().get_dump(dump_format="json")
+        trees = [json.loads(d) for d in model_dump]
+
+        if self._is_classification:
+            if self._output_size == 1:
+                return self._assemble_bin_class_output(trees)
+            else:
+                return self._assemble_multi_class_output(trees)
+        else:
+            return self._assemble_single_output(trees, self._base_score)
```

Comment:
For some reason I am very sensitive to unnecessary nested blocks.

Maybe again make them flat:

```python
if not self._is_classification:
    return self._assemble_single_output(trees, self._base_score)

if self._output_size == 1:
    return self._assemble_bin_class_output(trees)

return self._assemble_multi_class_output(trees)

```

Reply:
Well, I have an exact opposite sensitivity 😆 Not a big fun of premature return statement which are not last statements in the flow. This leads to messy return points in the code which are hard to keep track of. Seems like writing code in expressions only had irreversible impact on me.

Reply:
On the other hand branching gives you expression-like code with clearly defined logic flow.